hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -299,23 +300,24 @@ public synchronized void start() {
 
         heartbeat.start();
         connectToCluster();
-        if (isSmartRoutingEnabled) {
-            executor.scheduleWithFixedDelay(connectToAllClusterMembersTask, 1, 1, TimeUnit.SECONDS);
-        }
     }
 
-    public void connectToAllClusterMembers() {
+    public void tryConnectToAllClusterMembers(boolean sync) {
         if (!isSmartRoutingEnabled) {
             return;
         }
 
-        for (Member member : client.getClientClusterService().getMemberList()) {
-            try {
-                getOrConnect(member.getAddress());
-            } catch (Exception e) {
-                EmptyStatement.ignore(e);
+        if (sync) {
+            for (Member member : client.getClientClusterService().getMemberList()) {
+                try {
+                    getOrConnectToMember(member);
+                } catch (Exception e) {
+                    EmptyStatement.ignore(e);
+                }
             }
         }
+
+        executor.scheduleWithFixedDelay(connectToAllClusterMembersTask, 1, 1, TimeUnit.SECONDS);
```

Comment:
can `tryConnectToAllClusterMembers ` be called multiple times? if so, multiple scheduled tasks?

Reply:
No, it can not. We are calling it only once at the beginning of an instance.